lib/rules/prefer-template.js

```diff
@@ -187,6 +201,14 @@ module.exports = {
                 return sourceCode.getText(currentNode);
             }
 
+            if (isConcatenation(currentNode) && !hasIdentifierReference(currentNode)) {
+                const nodeTexts = sourceCode.getText(currentNode).split("+");
+
+                nodeTexts[nodeTexts.length - 1] = ` \`${nodeTexts[nodeTexts.length - 1].trim().slice(1)}`;
+
+                return nodeTexts.join("+");
```

Comment:
This can find `+` characters inside string literals:

```js
"a" + "b + c" + foo
```

Autofixed to:

```js
"a" + "b + `${  foo}`
```

If we need the rightmost node, it seems better to find it from the AST than to manually parse raw source code here.